helpers/apolloClient.ts

```diff
@@ -12,6 +12,25 @@ export const APOLLO_STATE_PROP_NAME = '__APOLLO_STATE__'
 let apolloClient: ApolloClient<NormalizedCacheObject>
 const cache = new InMemoryCache()
 
+import { persistCache } from 'apollo3-cache-persist'
+//restoring cache doesn't work with loading spinner, so I need to enable cache page by page
+const whiteList = ['/curriculum']
+;(async function () {
```

Comment:
This strange self-invoking function is the only way to do it. There is no global await and I can't move this logic elsewhere because `MyApp` page can't be async (some Next.js restriction).

Reply:
should be in code comment